tests/playwright/sanity/modules/ai/assets/js/editor/components/prompt-history/index.test.ts

```diff
@@ -0,0 +1,306 @@
+				await route.fulfill( {
+					json: getHistoryMock,
+				} );
+			}
+
+			if ( requestPostData.includes( 'ai_delete_history_item' ) ) {
+				await route.fulfill( {
+					json: deleteHistoryMock,
+				} );
+			}
+		} );
+	};
+
+	test( 'Prompt History - Common', async ( { page }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo );
+
+		const editor = await wpAdmin.openNewPage();
+
+		await test.step( 'Modal can be opened and closed', async () => {
+			await editor.addWidget( 'text-editor' );
+
+			await mockRoute( page, { getHistoryMock: noDataMock } );
+
+			await openPromptHistory( page );
+
+			await expect( page.locator( EditorSelectors.ai.promptHistory.modal ).first() ).toBeVisible();
+
+			await closePromptHistory( page );
+
+			await expect( page.locator( EditorSelectors.ai.promptHistory.modal ).first() ).toBeHidden();
+		} );
+
+		await test.step( 'Shows a message when there is a free plan', async () => {
+			await editor.addWidget( 'text-editor' );
+
+			await mockRoute( page, { getHistoryMock: noPlanMock } );
+
+			await openPromptHistory( page );
+
+			await expect( page.getByTestId( EditorSelectors.ai.promptHistory.upgradeMessageFull ).first() ).toBeVisible();
+
+			await closePromptHistory( page );
+		} );
+
+		await test.step( 'Shows a message when there are no history items', async () => {
+			await editor.addWidget( 'text-editor' );
+
+			await mockRoute( page, { getHistoryMock: noDataMock } );
+
+			await openPromptHistory( page );
+
+			await expect( page.getByTestId( EditorSelectors.ai.promptHistory.noDataMessage ).first() ).toBeVisible();
+
+			await closePromptHistory( page );
+		} );
+
+		await test.step( 'Renders items from different periods correctly', async () => {
+			await editor.addWidget( 'text-editor' );
+
+			await mockRoute( page, { getHistoryMock: differentPeriodsDataMock } );
+
+			await openPromptHistory( page );
+
+			await expect( page.getByTestId( EditorSelectors.ai.promptHistory.period ) ).toHaveCount( 2 );
+
+			await expect( page.getByTestId( EditorSelectors.ai.promptHistory.item ) ).toHaveCount( 2 );
+
+			await closePromptHistory( page );
+		} );
+
+		await test.step( 'Renders upgrade ad if a user has less than 90 items limit', async () => {
+			await editor.addWidget( 'text-editor' );
+
+			await mockRoute( page, { getHistoryMock: thirtyDaysLimitDataMock } );
+
+			await openPromptHistory( page );
+
+			await expect( page.getByTestId( EditorSelectors.ai.promptHistory.upgradeMessageSmall ).first() ).toBeVisible();
+
+			await closePromptHistory( page );
+		} );
+
+		await test.step( 'Renders a fallback icon for an unknown action', async () => {
+			await editor.addWidget( 'text-editor' );
+
+			await mockRoute( page, { getHistoryMock: unknownActionDataMock } );
+
+			await openPromptHistory( page );
+
+			await expect( page.getByTestId( EditorSelectors.ai.promptHistory.fallbackIcon ) ).toHaveCount( 1 );
+
+			await closePromptHistory( page );
+		} );
```

Comment:
Why E2E? isn't it a unit test at the component level?

Reply:
It's an internal team agreement that was here before I joined. I've specifically asked how should we cover code with tests and got a confirmation that we're using Playwright.